ssl/statem/statem_lib.c

```diff
@@ -1049,8 +1049,8 @@ int ssl_choose_server_version(SSL *s, CLIENTHELLO_MSG *hello)
              * wheter to ignore versions <TLS1.2 in supported_versions. At the
              * moment we honour them if present. To be reviewed later
              */
-            if ((int)candidate_vers > s->client_version)
-                s->client_version = candidate_vers;
+            /*if ((int)candidate_vers > s->client_version)
```

Comment:
So the effect of this change is that the client's highest supported version from the extension is used to choose the server version, but this value is not recorded anywhere?

Can you explain a bit more why that is correct w.r.t. surrounding code? Does nothing else assume that `s->client_version` is the highest supported client version? Perhaps stick a comment in `ssl_locl.h` reminding us that this is the `legacy_version` field in TLS 1.3?